touchhttp/bundle.go

```diff
@@ -123,24 +128,28 @@ var (
 	}
 )
 
-// labelNames extracts the names from the given Labels map.  This function
-// performs a single allocation and returns two slices of names:  (1) extra
-// are the label names other than the reserved names that are curried for
-// a given server or client, and (2) full are all the label names, including
-// the reserved names.
-func labelNames(l prometheus.Labels) (extra, full []string, err error) {
-	full = make([]string, 0, len(l))
-	for n := range l {
-		if n == CodeLabel || n == MethodLabel {
-			err = ErrReservedLabelName
-			return
-		} else {
-			full = append(full, n)
+// labelNames takes a sequence of name/value pairs and converts that into
+// a slice of names and a prometheus.Labels which should be used to curry
+// the associated metric.
+func labelNames(lvs []string) (names []string, curry prometheus.Labels, err error) {
```

Comment:
I seen mix opinions about named returns, what's your take on when to use them or not?

I often see opinions suggesting to use them for more complex use cases like immediate defer statements that reference a returned var